packages/suite/src/hooks/wallet/trading/form/common/useTradingComposeTransaction.ts

```diff
@@ -154,7 +172,30 @@ export const useTradingComposeTransaction = <T extends TradingSellExchangeFormPr
             if (typeof setMaxOutputId === 'number' && composed.max && shouldUpdateMaxAmount) {
                 setShouldUpdateMaxAmount(false);
 
-                setValue(TRADING_FORM_OUTPUT_AMOUNT, composed.max, {
+                const feeInUnits = composed.fee
+                    ? subunitsToUnits({
+                          value: asAmountSubunit(new BigNumber(composed.fee)),
+                          symbol: account.symbol,
+                      })
+                    : undefined;
+
+                const output = values.outputs?.[setMaxOutputId];
+                const token = findToken(account.tokens, output?.token);
+
+                const maxValue = isNetworkReserveEnabled
+                    ? getCryptoMaxAmountWithReserve(
+                          account.symbol,
+                          token?.contract,
+                          account.formattedBalance,
+                          composed.max,
+                          feeInUnits?.toString() || '0',
+                          isNetworkReserveEnabled,
+                      )
+                    : composed.max;
+
+                setShowReserveBanner(maxValue !== composed.max);
```

Comment:
```suggestion
                setShowReserveBanner(isNetworkReserveEnabled);
```
would this work too?

Reply:
I believe this would display the reserve banner every time, even when the user enters an amount that is still within balance and reserve, which is what we don't want. We only want to display it when the amount is adjusted, and that is when the original value is not equal to the adjusted value.